core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -117,6 +127,31 @@ public static String getStringOrNull(String property, JsonNode node) {
     return builder.build();
   }
 
+  public static Map<String, String> getStringMapOrNull(String property, JsonNode node) {
+    if (!node.has(property)) {
+      return null;
+    }
+
+    JsonNode pNode = node.get(property);
+    if (pNode == null || pNode.isNull()) {
+      return null;
+    }
+
+    Preconditions.checkArgument(pNode.isObject(),
+        "Cannot parse %s from non-object value: %s", property, pNode);
+    return buildStringMapFromJsonNodeObject(pNode);
+  }
+
+  private static Map<String, String> buildStringMapFromJsonNodeObject(JsonNode pNode) {
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+    Iterator<String> fields = pNode.fieldNames();
+    while (fields.hasNext()) {
+      String field = fields.next();
+      builder.put(field, getString(field, pNode));
+    }
+    return builder.build();
```

Comment:
This function is written such that it can be used in `getStringMap` and the differences between `getStringMap` and `getStringMapOrNull` would be in precondition checks, but I chose not to make that change to the existing code for now.

Reply:
Removed this function as `getStringMapOrNull` wound up not being needed.